flink-metrics/flink-metrics-dropwizard/src/main/java/org/apache/flink/dropwizard/ScheduledDropwizardReporter.java

```diff
@@ -184,7 +188,10 @@ public void notifyOfRemovedMetric(Metric metric, String metricName, MetricGroup
 	@Override
 	public String filterCharacters(String metricName) {
 		char[] chars = null;
-		final int strLen = metricName.length();
+		if (metricName.length() > maxComponentLength) {
+			log.warn("The metric name component {} exceeded the {} characters length limit and was truncated.", metricName, maxComponentLength);
+		}
+		final int strLen = Math.min(metricName.length(), maxComponentLength);
```

Comment:
You actually don't need to call `Math.min()` anymore after you already checked the condition for the warning message. You could thus assign `strLen` yourself.